src/tests/Interop/COM/NETClients/IDispatch/Program.cs

```diff
@@ -215,6 +215,65 @@ System.Collections.Generic.IEnumerable<int> GetEnumerable(System.Collections.IEn
+
+            Console.WriteLine($"Calling {nameof(DispatchCoerceTesting.ReturnToManaged)} ...");
+            
+            // Supported types
+            VarEnum[] supportedTypes =
+            {
+                VarEnum.VT_I2,
+                VarEnum.VT_I4,
+                VarEnum.VT_R4,
+                VarEnum.VT_R8,
+                VarEnum.VT_CY,
+                VarEnum.VT_DATE,
+                VarEnum.VT_BSTR,
+                VarEnum.VT_ERROR,
+                VarEnum.VT_BOOL,
+                VarEnum.VT_DECIMAL,
+            };
+
+            foreach (VarEnum vt in supportedTypes)
+            {
+                Console.WriteLine($"Converting {vt} to int should be supported.");
+                int result = dispatchCoerceTesting.ReturnToManaged((short)vt);
+                Assert.NotEqual(0, result);
+            }
+
+            // Invalid: Rejected before reaching coerce
+            Console.WriteLine("Invalid variant type should throw InvalidOleVariantTypeException.");
+            var variantException = Assert.Throws<InvalidOleVariantTypeException>(() => dispatchCoerceTesting.ReturnToManaged(0x7FFF));
+            Assert.Equal(unchecked((int)0x80131531), variantException.HResult);
+
+            // Not supported source or destination type: COMException { HResult: 0x80020005 }
+
+            // DISP_E_PARAMNOTFOUND: Converts to Missing
+            Console.WriteLine("Converting from VT_ERROR with DISP_E_PARAMNOTFOUND should be rejected.");
+            var comException = Assert.Throws<COMException>(() => dispatchCoerceTesting.ReturnToManaged(unchecked((short)((short)VarEnum.VT_ERROR | 0x8000))));
+            Assert.Equal(unchecked((int)0x80020005), comException.HResult);
+
+            Console.WriteLine("Converting int to VT_MISSING should be rejected.");
+            comException = Assert.Throws<COMException>(() => dispatchCoerceTesting.ReturnToManaged_Missing());
+            Assert.Equal(unchecked((int)0x80020005), comException.HResult);
+
+            Console.WriteLine("Converting int to VT_NULL should be rejected.");
+            comException = Assert.Throws<COMException>(() => dispatchCoerceTesting.ReturnToManaged_DBNull());
+            Assert.Equal(unchecked((int)0x80020005), comException.HResult);
+
+            // Rejected by VariantChangeTypeEx
+            Console.WriteLine("Converting VT_UNKNOWN to int should fail from VariantChangeTypeEx.");
+            Assert.Throws<InvalidCastException>(() => dispatchCoerceTesting.ReturnToManaged((short)VarEnum.VT_UNKNOWN));
+            Console.WriteLine("Converting VT_NULL to int should fail from VariantChangeTypeEx.");
+            Assert.Throws<InvalidCastException>(() => dispatchCoerceTesting.ReturnToManaged((short)VarEnum.VT_NULL));
+
+            // LOCAL_BOOL
+            Console.WriteLine("VARIANT_BOOL should convert to non-numeric string.");
+            Assert.NotEqual("-1", dispatchCoerceTesting.BoolToString());
```

Comment:
What should this convert to?

Reply:
It outputs "True" for me, but the doc says it may depend on current LCID.

> For conversions from VT_BOOL to VT_BSTR and back, uses the language specified by the locale in use on the local computer.

Reply:
We only operate with the empty LCID, so comparing against the case sensitive string is the more appropriate.

Reply:
Actually LCID of current culture gets used, passed from Binder. But VariantChangeTypeEx seems not giving different output for bool.

Reply:
All the runtime tests run on an English machine so that is the current culture and for all practical purpose is empty.